lib/cloudant.js

```diff
@@ -151,17 +151,27 @@ Cloudant.prototype.create = function(model, data, options, cb) {
   var self = this;
   var idName = self.idName(model);
   var mo = self.selectModel(model);
-  mo.db.insert(self.toDB(model, mo, data), function(err, result) {
-    debug('Cloudant.prototype.create insert %j %j', err, result);
-    if (err) {
-      if (err.statusCode === 409) {
-        err.message = err.message + ' (duplicate?)';
+  if (options.upsert === false) {
+    self.findById(model, data.id, {}, function(err) {
```

Comment:
@bajtos I get the idea from mongodb:

`MongoDB.prototype.replaceById` uses an option `upsert` to decide check id existed or not, see:
https://github.com/strongloop/loopback-connector-mongodb/blob/master/lib/mongodb.js#L962

But true, since Cloudant.prototype.create is a public method used by DAO it's dangers to expose that option...